modules/op-interacting-with-pipelines-using-the-developer-perspective.adoc

```diff
@@ -16,28 +16,38 @@ The *Pipelines* view in the *Developer* perspective lists all the pipelines in a
 +
 .Pipeline details
 image::op-pipeline-details.png[Pipeline details]
 +
-. Optionally, in the *Pipeline details* page:
-* Click the *Metrics* tab to see the following information about pipelines:
+. Optional: In the *Pipeline details* page, click the *Metrics* tab to see the following information about pipelines:
 ** *Pipeline Success Ratio*
 ** *Number of Pipeline Runs*
 ** *Pipeline Run Duration*
 ** *Task Run Duration*
 +
 You can use this information to improve the pipeline workflow and eliminate issues early in the pipeline lifecycle.
 +
-* Click the *YAML* tab to edit the YAML file for the pipeline.
-* Click the *Pipeline Runs* tab to see the completed, running, or failed runs for the pipeline.
+. Optional: Click the *YAML* tab to edit the YAML file for the pipeline.
+. Optional: Click the *Pipeline Runs* tab to see the completed, running, or failed runs for the pipeline.
++
+The *Pipeline Runs* tab provides information about the pipeline run and its details, the status of the task, and a link to debug failed pipeline runs. You can use the Options menu {kebab} to stop a running pipeline, to rerun a pipeline using the same parameters and resources as that of the previous pipeline execution, or to delete a pipeline run.
++
+* Click the required pipeline run to see the *Pipeline Run details* page. By default, the *Details* tab opens and provides a visual representation of all the serial tasks, parallel tasks, and the final tasks, along with the `when` expressions, if any, in the PipelineRun. Only if a pipeline run is successful, the results are displayed under the *Pipeline Run results* pane at the bottom of the page.
 +
 [NOTE]
 ====
 The *Details* section of the *Pipeline Run Details* page displays a *Log Snippet* of the failed pipeline run. *Log Snippet* provides a general error message and a snippet of the log. A link to the *Logs* section provides quick access to the details about the failed run.
-The *Log Snippet* is also displayed in the *Details* section of the *Task Run Details* page.
 ====
-You can use the Options menu {kebab} to stop a running pipeline, to rerun a pipeline using the same parameters and resources as that of the previous pipeline execution, or to delete a pipeline run.
-* Click the *Parameters* tab to see the parameters defined in the pipeline. You can also add or edit additional parameters, as required.
-* Click the *Resources* tab to see the resources defined in the pipeline. You can also add or edit additional resources, as required.
+* In the *Pipeline Run details* page, click the *Task Runs* tab to see the completed, running, or failed runs for the task.
```

Comment:
and not or. I am guessing it displays any of the states, not picking and choosing one.